src/app/TopicDasboard/ResourceEditForm.js

```diff
@@ -2,22 +2,45 @@ import React from 'react';
 import PropTypes from 'prop-types';
 
 //TODO:remove console.logs
-export function ResourceEditForm({ resource, handleUpdate, ref }) {
-  return (
-    <div>
-      <form
-        resourceid={resource.id}
-        className="resource-item"
-        onSubmit={handleUpdate}
-      >
-        <input type="text" name={resource.title} />
-        <button type="submit" className="resource-item-edit">
-          update
-        </button>
-      </form>
-      <a href={resource.uri}>{resource.uri}</a>
-    </div>
-  );
+export class ResourceEditForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.value = React.createRef();
+    this.state = {
+      value: this.props.resource.title,
+      uri: this.props.resource.uri
+    };
+  }
+
+  render() {
+    const { resource, handleUpdate } = this.props;
+    console.log(resource, 'uri');
+    return (
+      <div>
+        <form
```

Comment:
It feels pretty unnecessary to have the form submission be handled by a parent, and I think it's adding a lot of unnecessary complexity to how you're managing the inputs.

IMO the most idiomatic way to handle this would be:

1. Have the form handler within this form component

2. Create refs to each input like this:
```js
<input ref={input => this.SomeInputNameHere = input} />
```

3. Access the input values in the form event handler like this.
```js
handleSubmit = e => {
  const someNamedUserInput = this.someInputNameHere.value;
}
```

Reply:
Will need to come back and refactor this